### ISSUE TITLE
Allow passing config to jsdom.env

### ISSUE THREAD

**Issue by bfred-it** [lukechilds/node-browser-environment#1]
Example with jsdom:

```js
var jsdom = require("jsdom");

jsdom.env({
  userAgent: 'Some user agent'
});
```

Possible solution
```js

require('node-browser-environment')({
  // props: ['window', 'document', 'navigator'],
  userAgent: 'Some user agent'
});
```

**Comment by lukechilds** [comment]
@bfred-it Nice idea, I think it would be better to use a separate object for the `jsdom` config rather than mixing it in with our own though.

We could either have:

```js
require('node-browser-environment')(['window'], true, { userAgent: 'Some user agent' });
```

Which would keep current API compatibility. Or:

```js
require('node-browser-environment')({
  properties: ['window'],
  overwrite: true,
  jsdomConfig: {
    userAgent: 'Some user agent'
  }
});
```

Which is more a bit more readable but also much more verbose and is backwards incompatible. We also then loose the super simple syntax of:

```js
require('node-browser-environment')(['window']);
```

Which I really like and I think most people will use. Thoughts?

**Comment by bfred-it** [comment]
You can support both signatures:

* `nodeBrowserEnviroment(<array> props[, <bool> override])`
* `nodeBrowserEnviroment(<object> config)`

**Comment by lukechilds** [comment]
Yeah good point. Tbh a breaking change might not be a bad thing, I don't really like how you need to specify props to be able to set the second two args.

I might change it to just an array of props or a config object. That way we keep the simple array syntax and more complex use cases can use the config object. Sound good?

**Comment by bfred-it** [comment]
Sounds good to me 👍

**Comment by lukechilds** [comment]
Ok, I decided to remove the overwrite option, I'm checking any conflicts between the `jsdom` window properties and global node properties when the module is loaded and blacklisting them to stop weird bugs. This means all the other properties can be safely overwritten every time so there's no need for an option to opt in.

I've replaced the overwrite arg with the `jsdom` config arg so it can be used like this:

```js
require('node-browser-environment')(['window'], { userAgent: 'My User Agent' });
```

It can also be used on it's own without the properties arg:

```js
require('node-browser-environment')({ userAgent: 'My User Agent' });
```

It's a breaking API change so I've published as v2. You'll need to manually bump the version in your `package.json` to get the new functionality.
